code/__DEFINES/is_helpers.dm

```diff
@@ -7,6 +7,8 @@
 
 #define isatom(A) (isloc(A))
 
+#define isproperdatum(thing) (istype(thing, /datum) || istype(thing, /client))
+
```

Comment:
clients are childs of datums in our codebase because of black magic inside

Reply:
Do you have any idea why the proc existed in the first place? just like, a holdover from old code?

Reply:
likely, or maybe from a port of somebody who hadn't pulled our blackmagic change to clients down.

I find it funny because there is or was also "isref", an old proc we had that boiled down to "is thing a valid reference type" but was basically isdatum + isclient but it also had ismob and isicon and other stuff like that, that was already covered by isdatum (except isicon also returns true for cache references that are not datum)